nixos/modules/services/web-apps/pingvin-share.nix

```diff
@@ -0,0 +1,208 @@
+{ config, pkgs, lib, ... }:
+
+let
+  cfg = config.services.pingvin-share;
+  inherit (lib) mkOption mdDoc mkEnableOption mkIf mkPackageOptionMD types;
+in
+
+rec {
+  options = {
+    services.pingvin-share = {
+      enable = mkEnableOption (mdDoc "Pingvin Share, a self-hosted file sharing platform.");
+
+      user = mkOption {
+        type = types.str;
+        default = "pingvin";
+        description = mdDoc ''
+          User account under which Pingvin Share runs.
+        '';
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "pingvin";
+        description = mdDoc ''
+          Group under which Pingvin Share runs.
+        '';
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = mdDoc ''
+          Whether to open the firewall for the port in {option}`services.pingvin-share.frontend.port`.
+        '';
+      };
+
+      settings = {
```

Comment:
Isn't the `settings` key usually used for generating configuration through `pkgs.formats`, aka [RFC 42](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md)?

Reply:
I'm not sure that `settings` is only "restricted" for `pkgs.formats` but if you think that this should change, I can see two ways of doing so:
- rename `settings` to `config`
- delete `settings` and move everything setting to the upper level

What is the more suited option for you ?

Reply:
Yeah I don't really see any official guidance on this, but doing a quick search through the source, almost all `settings` are based on `pkgs.formats` and that's been my experience using NixOS as well.

Personally I would move these to the upper level. But I don't really have a strong opinion on this either. Maybe @SuperSandro2000 can chime in.

Reply:
I checked other services and indeed `settings` (or `config`) does not seem necessary at all so I moved these options to the upper level

Reply:
settings should be rfc42 style freeform settings.

Reply:
Indeed, that's why I moved every option one level up